core/iwasm/libraries/debug-engine/handler.c

```diff
@@ -441,14 +438,31 @@ handle_v_packet(WASMGDBServer *server, char *payload)
 void
 handle_threadstop_request(WASMGDBServer *server, char *payload)
 {
-    korp_tid tid = wasm_debug_instance_get_tid(
-        (WASMDebugInstance *)server->thread->debug_instance);
+    korp_tid tid;
     uint32 status;
+    WASMDebugInstance *debug_inst =
+        (WASMDebugInstance *)server->thread->debug_instance;
+    bh_assert(debug_inst);
+
+    /* According to
+       https://sourceware.org/gdb/onlinedocs/gdb/Packets.html#Packets, the "?"
+       package should be sent when connection is first established to query the
+       reason the target halted */
+    bh_assert(debug_inst->current_state == DBG_LAUNCHING);
 
-    tid = wasm_debug_instance_wait_thread(
-        (WASMDebugInstance *)server->thread->debug_instance, tid, &status);
+    /* Waiting for the stop event */
+    while (!debug_inst->stopped_thread) {
```

Comment:
Dead loop occupies a lot CPU resource, is it possible to wait a signal? Or add a timeout wait inside the loop?

Reply:
Done